x-pack/functionbeat/manager/gcp/cli_manager.go

```diff
@@ -0,0 +1,169 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/x-pack/functionbeat/function/provider"
+	"github.com/elastic/beats/x-pack/functionbeat/manager/executor"
+	fngcp "github.com/elastic/beats/x-pack/functionbeat/provider/gcp/gcp"
+)
+
+type installer interface {
+	Config() *fngcp.FunctionConfig
+	Name() string
+}
+
+// CLIManager interacts with the AWS Lambda API to deploy, update or remove a function.
+// It will take care of creating the main lambda function and ask for each function type for the
+// operation that need to be executed to connect the lambda to the triggers.
+type CLIManager struct {
+	templateBuilder *defaultTemplateBuilder
+	location        string
+	tokenSrc        oauth2.TokenSource
+	log             *logp.Logger
+	config          *Config
+}
+
+// Deploy delegates deploy to the actual function implementation.
+func (c *CLIManager) Deploy(name string) error {
+	c.log.Debugf("Deploying function: %s", name)
+	defer c.log.Debugf("Deploy finish for function '%s'", name)
+
+	create := false
```

Comment:
This line might be a little confusing. Let's just pass the bool value to `c.deploy`. If you want to make the call to deploy more readable rather introduce an enum like:

```
type deployMode int

const (
  deployCreate deployMode = iota
  deployUpdate
)
```